crates/accelerate/src/sabre_swap/mod.rs

```diff
@@ -383,6 +383,7 @@ fn swap_map_trial(
     // Main logic loop; the front layer only becomes empty when all nodes have been routed.  At
     // each iteration of this loop, we route either one or two gates.
     let mut routable_nodes = Vec::<NodeIndex>::with_capacity(2);
+    let mut swap_scratch = Vec::<[VirtualQubit; 2]>::new();
```

Comment:
This is self explanatory now, but do you think a comment on the scope and use of this would be useful? Just to say this is a reused vec that tracks swaps as we iterate and doesn't shrink so we save allocations. It's a bit of an unusual pattern unless you know what it's being used for.

Reply:
Done in 422964e42.